packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,351 @@
+    AssociationManager.associateTagOption(this, this.portfolioId, tagOption);
+  }
+
+  /**
+   * Create a unique id based off the L1 portfolio.
+   */
+  protected abstract getKeyForPortfolio(value: string): string;
+}
+
+/**
+ * Properties for a Portfolio.
+ */
+export interface PortfolioProps {
+
+  /**
+   * Enforces a particular physical portfolio name.
+   */
+  readonly portfolioName: string;
+
+  /**
+   * The provider name.
+   */
+  readonly providerName: string;
+
+  /**
+   * The accept language.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * Description for portfolio.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * A collection of tags attached to portfolio.
+   * @default - No tags provided
+   */
+  readonly tags?: Tag[];
+}
+
+/**
+ * A Service Catalog portfolio.
+ */
+export class Portfolio extends PortfolioBase {
+
+  /**
+   * Creates a Portfolio construct that represents an external portfolio.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs Portfolio import properties
+   */
+  public static fromPortfolioAttributes(scope: Construct, id: string, attrs: PortfolioAttributes): IPortfolio {
+    const parts = Stack.of(scope).parseArn(attrs.portfolioArn);
+    function throwError(errorMessage: string): never {
+      throw new Error(errorMessage);
+    }
+    class Import extends PortfolioBase {
+      public readonly id = id
+      public readonly portfolioId = parts.resourceName ?? throwError('Portfolio arn missing Portfolio ID during import from attributes');
+      public readonly portfolioArn = attrs.portfolioArn;
+      public readonly portfolioName = attrs.portfolioName;
+
+      protected getKeyForPortfolio(value: string): string {
+        return getIdentifier(this.portfolioArn, value);
+      }
+    }
+    return new Import(scope, id);
+  }
+
+  public readonly portfolioArn: string;
+  public readonly portfolioName: string;
+  public readonly portfolioId: string;
+
+  private readonly portfolio: CfnPortfolio;
+
+  constructor(scope: Construct, id: string, props: PortfolioProps) {
+    super(scope, id, {
+      physicalName: props.portfolioName,
+    });
+
+    this.validatePortfolioProps(props);
+
+    this.portfolio = new CfnPortfolio(this, 'Resource', {
+      displayName: props.portfolioName,
+      providerName: props.providerName,
+      description: props.description,
+      tags: props.tags,
+      acceptLanguage: props.acceptLanguage,
+    });
+
+    this.portfolioArn = this.getResourceArnAttribute(this.portfolio.ref, {
+      service: 'catalog',
+      resource: 'portfolio',
+      resourceName: this.physicalName,
+    });
+
+    this.portfolioName = this.getResourceNameAttribute(props.portfolioName);
+    this.portfolioId = this.getResourceNameAttribute(this.portfolio.ref);
+  }
```

Comment:
I'm a little confused with all of the attributes. So the `portfolioName` is _not_ what goes into the ARN? Is it `portfolioId` instead, and is it automatically generated upon creating a Portfolio?